test/iterators.jl

```diff
@@ -914,3 +914,34 @@ end
     @test !Base.isdone(gen)
     @test collect(gen) == ["foo"]
 end
+
+@testset "Unfold" begin
+    @test isempty(Iterators.Unfold(identity, nothing))
+    @test all(2 .^ (0:61) .== Iterators.Unfold(x -> x < 2^62 ? (x, (x+x)) : nothing, 1))
```

Comment:
```suggestion
    unfold61 = Iterators.Unfold(1) do x
         if x < 2^26
            return x, x+x
         else
            return nothing
         end
    end
    @test all(enumerate(unfold61)) do (i,x)
        2^(i-1) == x
    end
```

Easier to read & no intermediate allocation due to another iterator. Otherwise an array has to be created due to the broadcasting before it's passed to `all`.